### PR TITLE
Move selinux label reservations to containers storage.

### PR DESCRIPTION
**Pull request by rhatdan**

### CODE REVIEW

File: libpod/container_internal.go [hunk 1]
```diff
@@ -194,10 +194,16 @@ func (c *Container) setupStorage(ctx context.Context) error {
...
 	}
 
-	var options *storage.ContainerOptions
-	if c.config.Rootfs == "" {
-		options = &storage.ContainerOptions{c.config.IDMappings}
+	options := storage.ContainerOptions{
+		IDMappingOptions: storage.IDMappingOptions{
+			HostUIDMapping: true,
+			HostGIDMapping: true,
+		},
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply libpod/container_internal.go]: I thought it was.  It was set inside of the calling function if the coptions was nil.

**Comment by baude** [comment]
please rebase

**Comment by baude** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
bot, retest this please

File: libpod/storage.go [hunk 2]
```diff
@@ -59,7 +62,7 @@ func (metadata *RuntimeContainerMetadata) SetMountLabel(mountLabel string) {
 
 // CreateContainerStorage creates the storage end of things.  We already have the container spec created
 // TO-DO We should be passing in an Image object in the future.
-func (r *storageService) CreateContainerStorage(ctx context.Context, systemContext *types.SystemContext, imageName, imageID, containerName, containerID, mountLabel string, options *storage.ContainerOptions) (cinfo ContainerInfo, err error) {
+func (r *storageService) CreateContainerStorage(ctx context.Context, systemContext *types.SystemContext, imageName, imageID, containerName, containerID, mountLabel string, options storage.ContainerOptions) (cinfo ContainerInfo, err error) {
```
Comment (mheon) [inline libpod/storage.go]: Do we still want mountLabel in the arguments here, since it's being assigned by c/storage now?

**Comment by mheon** [comment]
One little thing, otherwise looks good

**Comment by baude** [comment]
bot, retest this please

File: libpod/storage.go (hunk 2 above)
Comment (rhatdan) [reply libpod/storage.go]: Good Catch. This caused me to find another issue with Inspect.  Fixed both.

**Comment by mheon** [comment]
CI says you have lint errors, otherwise looks fine

**Comment by rhatdan** [comment]
These failures are caused by differences between Golang 1.10 lint and golang 1.11 lint.

File: vendor.conf [hunk 3]
```diff
@@ -11,7 +11,7 @@ github.com/containerd/continuity master
 github.com/containernetworking/cni v0.7.0-alpha1
 github.com/containernetworking/plugins 1562a1e60ed101aacc5e08ed9dbeba8e9f3d4ec1
 github.com/containers/image bd10b1b53b2976f215b3f2f848fb8e7cad779aeb
-github.com/containers/storage 41294c85d97bef688e18f710402895dbecde3308
+github.com/containers/storage 24f0de45708bc6e4c8062828cd03812aaebc30db https://github.com/rhatdan/storage
```
Comment (giuseppe) [inline vendor.conf]: should containers/storage come from rhatdan/storage?

Comment (rhatdan) [reply vendor.conf]: Nope that is a bug.

Comment (rhatdan) [reply vendor.conf]: Although I don't see this in the master branch.

**Pull request merged**